### PR TITLE
Fix #1956 Division by zero FPD

### LINKED ISSUE #1956: Division by zero FPD
All found issues originate from the maxwarningpoints setting being equal to zero and emit a PHP warning Division by zero.

https://github.com/mybb/mybb/blob/feature/admin/modules/user/users.php#L1056
https://github.com/mybb/mybb/blob/feature/admin/modules/user/users.php#L3570

https://github.com/mybb/mybb/blob/feature/inc/datahandlers/warnings.php#L373
https://github.com/mybb/mybb/blob/feature/inc/datahandlers/warnings.php#L374
https://github.com/mybb/mybb/blob/feature/inc/datahandlers/warnings.php#L561
https://github.com/mybb/mybb/blob/feature/inc/datahandlers/warnings.php#L562

https://github.com/mybb/mybb/blob/feature/inc/functions_post.php#L377

https://github.com/mybb/mybb/blob/feature/member.php#L2430

https://github.com/mybb/mybb/blob/feature/usercp.php#L3694

https://github.com/mybb/mybb/blob/feature/warnings.php#L344
https://github.com/mybb/mybb/blob/feature/warnings.php#L411
https://github.com/mybb/mybb/blob/feature/warnings.php#L741

repsperpage setting-related:
https://github.com/mybb/mybb/blob/feature/reputation.php#L835
https://github.com/mybb/mybb/blob/feature/reputation.php#L854 (zero as an argument)

membersperpage setting-related:
https://github.com/mybb/mybb/blob/feature/managegroup.php#L355 (zero as an argument)

Original thread: [Division by zero FPD](http://community.mybb.com/thread-169573.html)

### PR DESCRIPTION
**Pull request by Devilshakerz**

### CODE REVIEW

File: managegroup.php [hunk 1]
```diff
@@ -341,7 +341,13 @@
 	{
```
Comment (Sama34) [inline managegroup.php]: Can you remove this commented _if_?

File: managegroup.php [hunk 2]
```diff
@@ -341,7 +341,13 @@
...
 		error($lang->group_no_members);
 	}*/
+	
 	$perpage = $mybb->settings['membersperpage'];
```
Comment (Sama34) [inline managegroup.php]: Can you cast the setting value here to `(int)`?

File: admin/modules/user/users.php [hunk 3]
```diff
@@ -1053,6 +1053,11 @@
 
 	if($mybb->settings['enablewarningsystem'] != 0 && $user_permissions['canreceivewarnings'] != 0)
 	{
+		if (!$mybb->settings['maxwarningpoints'])
```
Comment (Destroy666x) [inline admin/modules/user/users.php]: There should be no space between `if` and `(` to be consistent with our standards

File: managegroup.php (hunk 2 above)
Comment (Destroy666x) [inline managegroup.php]: It could use the `< 1` check too.

Comment (Devilshakerz) [inline managegroup.php]: I don't see any issues with negative numbers here, but i can convert all the checks to `< 1`.

Comment (Destroy666x) [inline managegroup.php]: Oh, my bad, I thought it was memberlist.php, which throws an SQL error on the profile fields query with negative `$mybb->settings['membersperpage']`. No reason not to use the `< 1` check nonetheless.

File: reputation.php [hunk 4]
```diff
@@ -827,12 +827,19 @@
...
+	{
+		$perpage = 15;
+	}
+
 	// Check if we're browsing a specific page of results
 	if($mybb->get_input('page', MyBB::INPUT_INT) > 0)
 	{
 		$page = $mybb->get_input('page', MyBB::INPUT_INT);
-		$start = ($page-1) * $mybb->settings['repsperpage'];
 		$pages = $reputation_count / $mybb->settings['repsperpage'];
```
Comment (Destroy666x) [inline reputation.php]: Any reason you left this line? It will sitll produce FPD. Except that the PR looks ready to merge.

**Pull request merged**